rust/nasl-builtin-cert/src/lib.rs

```diff
@@ -0,0 +1,441 @@
+        "1.2.840.113549.1.1.4" => "md5WithRSAEncryption".to_string(),
+        "1.2.840.113549.1.1.6" => "rsaOAEPEncryptionSET".to_string(),
+        "1.2.840.10045.3.1.1" => "secp192r1".to_string(),
+        "1.3.132.0.1" => "sect163k1".to_string(),
+        "1.3.132.0.15" => "sect163r2".to_string(),
+        "1.3.132.0.33" => "secp224r1".to_string(),
+        "1.3.132.0.26" => "sect233k1".to_string(),
+        "1.3.132.0.27" => "sect233r1".to_string(),
+        "1.2.840.10045.3.1.7" => "secp256r1".to_string(),
+        "1.3.132.0.16" => "sect283k1".to_string(),
+        "1.3.132.0.17" => "sect283r1".to_string(),
+        "1.3.132.0.34" => "secp384r1".to_string(),
+        "1.3.132.0.36" => "sect409k1".to_string(),
+        "1.3.132.0.37" => "sect409r1".to_string(),
+        "1.3.132.0.35" => "sect521r1".to_string(),
+        "1.3.132.0.38" => "sect571k1".to_string(),
+        "1.3.132.0.39" => "sect571r1".to_string(),
+        "2.16.840.1.101.3.4.3.1" => "id-dsa-with-sha224".to_string(),
+        "2.16.840.1.101.3.4.3.2" => "id-dsa-with-sha256".to_string(),
+        "2.16.840.1.101.3.4.2.1" => "sha256".to_string(),
+        "2.16.840.1.101.3.4.2.2" => "sha384".to_string(),
+        "2.16.840.1.101.3.4.2.3" => "sha512".to_string(),
+        "2.16.840.1.101.3.4.2.4" => "sha224".to_string(),
+        _ => "unknown".to_string(),
+    }
+}
+
+fn pub_key_alg_oid_to_name(name: &str) -> String {
+    match name {
+        "1.2.840.113549.1.1.1" => "RSA".to_string(),
+        "2.5.8.1.1" => "RSA (X.509)".to_string(),
+        "1.2.840.113549.1.1.4" => "RSA (MD5)".to_string(),
+        "1.2.840.113549.1.1.5" => "RSA (SHA1)".to_string(),
+        "1.2.840.10040.4.1" => "DSA".to_string(),
+        "1.2.643.2.2.19" => "GOST R 34.10-2001".to_string(),
+        "1.2.643.2.2.20" => "GOST R 34.10-94".to_string(),
+        "1.2.840.10045.2.1" => "EC".to_string(),
+        _ => "unknown".to_string(),
+    }
+}
+
+fn subject_oid_to_name(oid: &str) -> String {
+    match oid {
+        "2.5.4.6" => "C".to_string(),
+        "2.5.4.8" => "ST".to_string(),
+        "2.5.4.7" => "L".to_string(),
+        "2.5.4.10" => "O".to_string(),
+        "2.5.4.3" => "CN".to_string(),
+        "2.5.4.11" => "OU".to_string(),
+        "2.5.4.12" => "T".to_string(),
+        "2.5.4.42" => "GN".to_string(),
+        "2.5.4.43" => "I".to_string(),
+        "2.5.4.4" => "SN".to_string(),
+        _ => oid.to_string(),
+    }
+}
+
+impl NaslCerts {
+    /// Create a certificate object.
+    ///
+    /// Takes a string/data as unnamed argument and returns an identifier
+    /// used with the other cert functions. The data is usually the BER
+    /// encoded certificate but the function will also try a PEM encoding
+    /// on failure to parse BER encoded one.
+    ///
+    /// On success the function returns a cert identifier that can be used
+    /// for further operations.
+    #[nasl_function]
+    fn cert_open(self, cert: &[u8]) -> Result<usize, FunctionErrorKind> {
```

Comment:
Currently we have this issue in many cases, as for the c implementation string an data have different names, but are handled the same internally. I guess this issue here will be fixed, after `SC-1148`
I would not write a todo, as this will be fixed without touching this code in any way (I guess)